modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2665,37 +2686,11 @@ private CompletableFuture<CompletableFuture<?>> applyUpdateAllCommand(
 
             if (!cmd.full()) {
                 if (skipDelayedAck) {
-                    // TODO: https://issues.apache.org/jira/browse/IGNITE-20124 Temporary code below
-                    synchronized (safeTime) {
-                        storageUpdateHandler.handleUpdateAll(
-                                cmd.txId(),
-                                cmd.rowsToUpdate(),
-                                cmd.tablePartitionId().asTablePartitionId(),
-                                true,
-                                null,
-                                null,
-                                indexIdsAtRwTxBeginTs(transactionId)
-                        );
-
-                        updateTrackerIgnoringTrackerClosedException(safeTime, cmd.safeTime());
-                    }
+                    handleUpdateAllUnderLocksOnSnapshotsAndSafeTime(transactionId, cmd);
```

Comment:
I don't think that all that private methods actually increase readability. Matter of taste though.

Reply:
They decrease duplication, hence they decrease a change for an error